tests/ui/methods_unfixable.rs

```diff
@@ -0,0 +1,29 @@
+#![warn(clippy::filter_next)]
+
+fn main() {
+    issue10029::puzzle("");
+}
+
+mod issue10029 {
```

Comment:
We can probably make this testcase simpler

Reply:
Reducing it is great since it makes us understand the problem more (not in this case though), but also makes it simpler for others to understand what's happening and what is being tested in case anyone looks at this in the future :smile:

Reply:
Ok, fair points, I reduced the test case down to the core problem in the last commit👍